src/sage/combinat/symmetric_group_algebra.py

```diff
@@ -3065,24 +3065,34 @@ def q(self):
         """
         return self._q
 
-    def _coerce_start(self, x):
+    def _element_constructor_(self, x):
         """
         EXAMPLES::
 
             sage: H3 = HeckeAlgebraSymmetricGroupT(QQ, 3)
-            sage: H3._coerce_start([2,1])
+            sage: H3([2,1])  # indirect doc test
+            T[2, 1, 3]
+            sage: H3( SymmetricGroup(3).an_element() )
+            [1, 3, 2]
+            sage: H3( [2, 1] )
+            T[2, 1, 3]
+            sage: H3( [2, 1, 3, 4, 5, 6] )
             T[2, 1, 3]
         """
```

Comment:
Maybe add examples for both `SymmetricGroup(3).an_element()` and `Permutations(3).an_element()` to match the description in this PR?